src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+// starts with a verb and concisely describes what it does.
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  const newName = nameInput.value;
+  const newJob = jobInput.value;
+  profileSaveBtn.textContent = "Saving...";
+  api
+    .updateProfileInfo({ name: newName, about: newJob })
+    .then((data) => {
+      profileNameElement.textContent = data["name"];
+      profileJobElement.textContent = data["about"];
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      closeModal(editModal);
+      profileSaveBtn.textContent = "Save";
+    });
+}
+
+// Connect the handler to the form, so it will watch for the submit event.
+profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+
+const cardsList = document.querySelector(".cards__list");
+const cardTemplate = document.querySelector("#card");
+
+let selectedCard;
+let selectedCardId;
+
+function getCardElements(data) {
+  const cardElement = cardTemplate.content.cloneNode(true);
+  const likeButton = cardElement.querySelector(".card__button");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  const cardImage = data["link"];
+  const cardName = data["name"];
+  const cardImageElement = cardElement.querySelector(".card__image");
+
+  cardImageElement.src = cardImage;
+  cardImageElement.alt = cardName;
+  cardElement.querySelector(".card__name").textContent = cardName;
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(evt, data["_id"]);
+  });
+
+  if (data.isLiked) {
+    likeButton.classList.add("card__button_liked");
+  }
+
+  likeButton.addEventListener("click", (evt) => {
+    handleLike(evt, data["_id"]);
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    photoModalImage.src = cardImage;
+    photoModalImage.alt = cardName;
+    photoModalCaption.textContent = cardName;
+    openModal(photoModal);
+  });
+  return cardElement;
+}
+
+const postAddButton = document.querySelector(".profile__new-btn");
+const addModal = document.querySelector("#add-modal");
+
+const postFormElement = document.forms["add-post"];
+const postSaveBtn = postFormElement.querySelector(".modal__save-button");
+
+const linkInput = postFormElement.querySelector("#link");
+const captionInput = postFormElement.querySelector("#caption");
+
+postAddButton.addEventListener("click", () => {
+  openModal(addModal);
+  postSaveBtn.disabled = true;
+  postSaveBtn.classList.add("modal__save-button_disabled");
+});
+
+function handleAddFormSubmit(evt) {
+  evt.preventDefault();
+  const newCard = {
+    name: captionInput.value,
+    link: linkInput.value,
+  };
+
+  postSaveBtn.textContent = "Saving...";
+  api
+    .makeCard({ link: newCard.link, name: newCard.name })
+    .then((res) => {
+      cardsList.prepend(getCardElements(res));
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors